### PR TITLE
fix(#348): assertQueueErrorHandler initial impl

### PR DESCRIPTION
**Pull request by christopheblin**
provide an error handler to handle failure during queue creation (see #348)

### CODE REVIEW

**Comment by underfisk** [comment]
@christopheblin LGTM but there seems to be a change on the underlying amqlib package, do we really need that version update?

**Comment by christopheblin** [comment]
@underfisk 

you're talkkng about the yarn.lock ? 

inside the file package.json (https://github.com/golevelup/nestjs/blob/master/packages/rabbitmq/package.json#L41), you already have (this is not me here)
```
"amqplib": "^0.8.0",
```

I jut ran yarn install and it automatically upgraded the yarn.lock according to what is specified by package.json ...

**Comment by underfisk** [comment]
> 

Just wanted to be sure, long time i don't touch the package but I'll take a look at this PR soonish

File: packages/rabbitmq/src/amqp/connection.ts [hunk 1]
```diff
@@ -416,16 +417,25 @@ export class AmqpConnection {
...
     let actualQueue: string;
 
     if (createQueueIfNotExists) {
-      const { queue } = await channel.assertQueue(
-        subscriptionOptions.queue || '',
-        subscriptionOptions.queueOptions || undefined
-      );
-      actualQueue = queue;
+      const queueName = subscriptionOptions.queue || '';
+      const queueOptions = subscriptionOptions.queueOptions || undefined;
```
Comment (underfisk) [inline packages/rabbitmq/src/amqp/connection.ts]: If `queueOptions` is not defined its going to be undefined so using `||` operator is not necessary

File: packages/rabbitmq/src/amqp/connection.ts [hunk 2]
```diff
@@ -416,16 +417,25 @@ export class AmqpConnection {
...
 
     let actualQueue: string;
 
     if (createQueueIfNotExists) {
-      const { queue } = await channel.assertQueue(
-        subscriptionOptions.queue || '',
-        subscriptionOptions.queueOptions || undefined
-      );
-      actualQueue = queue;
+      const queueName = subscriptionOptions.queue || '';
```
Comment (underfisk) [inline packages/rabbitmq/src/amqp/connection.ts]: Use `??` operator instead of `||`

**Review by underfisk — CHANGES_REQUESTED** [review]
LGTM, address the changes and I'll merge this

File: packages/rabbitmq/src/amqp/connection.ts (hunk 1 above)
Comment (christopheblin) [reply packages/rabbitmq/src/amqp/connection.ts]: should be good

File: packages/rabbitmq/src/amqp/connection.ts (hunk 2 above)
Comment (christopheblin) [reply packages/rabbitmq/src/amqp/connection.ts]: should be good

**Comment by WonderPanda** [comment]
This has been published and is available in `@golevelup/nestjs-rabbitmq@1.20.0`

**Pull request merged**